### PR TITLE
python312Packages.gunicorn: 22.0.0 -> 23.0.0

### PR DESCRIPTION
**Pull request by getchoo**
## Description of changes

~~Fixes CVE-2024-1135~~ This was already patched last release, yet it's mentioned again in the changelog

Changelog: https://github.com/benoitc/gunicorn/releases/tag/23.0.0
Diff: https://github.com/benoitc/gunicorn/compare/22.0.0...23.0.0

Built on `nixos-unstable`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by risicle** [comment]
This breaks `mlflow`: `- gunicorn<23 not satisfied by version 23.0.0`

**Comment by getchoo** [comment]
> This breaks mlflow: - gunicorn<23 not satisfied by version 23.0.0

Thanks for catching this! Seems to be fixed now

**Comment by SuperSandro2000** [comment]
@ofborg build pretalx pretix

**Review by Atemu — APPROVED** [review]
Cherry-picked this onto unstable and deployed paperless on my machine which appears to work just fine.

**Comment by Atemu** [comment]
Built pretix and pretalx tests too. Anything else we should try?

**Review by risicle — APPROVED** [review]
Tested numerous dependent packages on macos 12 x86_64 & nixos x86_64

**Comment by SuperSandro2000** [comment]
@ofborg build python312Packages.pyarrow

File: pkgs/development/python-modules/gunicorn/default.nix [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 buildPythonPackage rec {
   pname = "gunicorn";
-  version = "22.0.0";
+  version = "23.0.0";
   pyproject = true;
 
   disabled = pythonOlder "3.6";
```
Comment (dotlambda) [inline pkgs/development/python-modules/gunicorn/default.nix:28]:
```suggestion
  disabled = pythonOlder "3.7";
```

**Review by SuperSandro2000 — APPROVED** [review]
there are other, unrelated breakages on staging which make assessing this a bit hard but should be fine I hope.

**Pull request merged**